cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/openai/OpenAIPrompt.scala

```diff
@@ -136,10 +180,20 @@ class OpenAIPrompt(override val uid: String) extends Transformer
     }
   }
 
-  override def transformSchema(schema: StructType): StructType =
-    openAICompletion
-      .transformSchema(schema)
-      .add(getPostProcessing, getParser.outputSchema)
+  override def transformSchema(schema: StructType): StructType = {
+    openAICompletion match {
+      case chatCompletion: OpenAIChatCompletion =>
+        chatCompletion.setMessagesCol("messages")
```

Comment:
had to do this as MessageCol didn't have a default value and was causing issues when calling transformSchema - let me know if there's a better workaround

Reply:
This might be because completion expects promtCol to be set where chat completion expects messagesCol to be set